frontend/src/Components/UserProfile.js

```diff
@@ -1,26 +1,24 @@
-import React, { useEffect, useState, useContext } from "react";
-import { AuthContext } from "../Context/AuthContext";
-import Post from "./Post/Post";
+import React, { useState, useEffect } from "react";
 import axios from "axios";
+import { GET_USER_DETAILS } from "../constants";
 import { Loading } from "../Components/Common/Loader";
-import { POST_GET_ALL_URL } from "../constants";
+import Post from "./Post/Post";
 
-export default function UserProfile() {
-  const { user, token } = useContext(AuthContext);
-  const [posts, setPosts] = useState([]);
+export default function UserProfile({ id }) {
+  const [user, setUser] = useState([]);
   const [loading, setLoading] = useState(false);
 
   useEffect(() => {
     setLoading(true);
     axios
-      .get(POST_GET_ALL_URL, {
-        headers: {
-          Authorization: "Bearer " + token,
-        },
-      })
+      .get(GET_USER_DETAILS + `${id}`)
       .then((res) => {
-        setPosts(res.data.data);
         setLoading(false);
+        setUser(res.data.data);
+      })
+      .catch((err) => {
+        setLoading(false);
+        console.log("User Not found"); //catch error in backend and update here
```

Comment:
We can simply `toast.error("User does not exist")`?

Reply:
Done